Software use case rewrite

yogeshbdeshpande opened the pull request.

rjb4standards commented:
@yogeshbdeshpande the revisions seem to place greater emphasis on package managers, removing the original intent. The original intent was much broader and could be applied to a wide swath of software distribution, i.e. GitHub, app stores, and package manager repos.

yogeshbdeshpande commented:
> @yogeshbdeshpande the revisions seem to place greater emphasis on package managers, removing the original intent. The original intent was much broader and could be applied to a wide swath of software distribution, i.e. GitHub, app stores, and package manager repos.

Sure, we can broaden the scope. Not a problem!

kaywilliams reviewed (commented):
Previously I had suggested focusing on lower-level use cases, but after reviewing the document with Yogesh and Henk, I believe the current approach works.

yogeshbdeshpande commented:
> should be on satisfying lower-level use case

We had a detail discussion about this between me and Henk. Basically, I was all for the support of going to depict a real world example for each of the situation. However, this makes the document excessively long and verbose. 

We should make a judgment as a team as to:

1. Where should the Use Case document reside ? Some members feel within Arch. Document, some feel as a separate RFC.
2. Given the number of scenarios, should we abstract it to generic use case or we depict a specific example, next to each of them ?
3. Is too long a document (when a separate RFC) acceptable to the group ?

yogeshbdeshpande commented on draft-birkholz-scitt-software-use-cases.md:
```diff
@@ -85,55 +85,53 @@ deployment chain diagram here
-* enable the consumer to verify that a trust bond for a certain software component package exists and is still valid.
+* enable the consumer to verify that a relationship link for a certain software component package  between the supplier and the distributor(its signing authority) exists and is still valid.
+
+## Multi party Evaluation of a Released Software Component
```
```suggestion
## Multi Stakeholder Evaluation of a Released Software Component
```

rjb4standards replied:
There is no requirement that the distributor be the signing authority. The signing authority is whomever the software supplier chooses, and many cases the supplier and signing authority are the same entity. That's how it is with our SAG-PM software; REA is both supplier and signing authority.

yogeshbdeshpande replied:
@rjb4standards are you available sometime tomorrow to have a discussion with me and Henk?
Also what is your email address? We want to schedule a meeting with you..

rjb4standards replied:
dick@reliableenergyanalytics.com   I can meet tomorrow

yogeshbdeshpande committed:
Revamping SW use case rewrite

yogeshbdeshpande commented:
@henkbirkholz @rjb4standards @kaywilliams I have redone the PR post our meeting today. 

Please have a look..

yogeshbdeshpande commented:
> > @yogeshbdeshpande the revisions seem to place greater emphasis on package managers, removing the original intent. The original intent was much broader and could be applied to a wide swath of software distribution, i.e. GitHub, app stores, and package manager repos.
> 
> Sure, we can broaden the scope. Not a problem!

This has been resolved now!

yogeshbdeshpande committed:
Few more comments

Signed-off-by: Yogesh Deshpande <yogesh.deshpande@arm.com>

OR13 commented on draft-birkholz-scitt-software-use-cases.md:
```diff
@@ -85,57 +85,51 @@ deployment chain diagram here
 # Software Supply Chain Use Cases
 
-## Trustworthiness Relationship between Product Supplier and the Distributor of the Product
+## Verifiable Authenticity in Software Distribution Systems
```
I like the section header change

kaywilliams replied:
+1

OR13 commented on draft-birkholz-scitt-software-use-cases.md:
```diff
@@ -244,35 +232,22 @@ Even with a robust firmware update system, the following problems remain as give
+* ISV attempts to show that it did all the steps correctly. It does disclose information about the binary they delivered. In addition, they also reveal their build environment and the architecture they used during the build.
 
-* The deadlock continues, with no clear resolution.
+* However, there are no "Verifiable Proofs" of the statement made by ISV. All the stakeholders, in the ecosystem (end user, car manufacturer and the integrator) has to trust without any ability to verify the claims made by the ISV.
```
I like the change.

OR13 reviewed (approved):
great improvements

kaywilliams committed:
Update draft-birkholz-scitt-software-use-cases.md

Edit the first use case to shift the focus to the software consumer at installation.
